File: backend/src/ollama.ts
import axios from 'axios';

const OLLAMA_BASE = process.env.OLLAMA_URL || 'http://localhost:11434';
const MODEL = 'gemma3:4b';
const TIMEOUT = 60000;

interface OllamaResponse {
  response: string;
}

export async function generate(prompt: string): Promise<string> {
  const res = await axios.post<OllamaResponse>(
    `${OLLAMA_BASE}/api/generate`,
    { model: MODEL, prompt, stream: false, options: { temperature: 0 } },
    { timeout: TIMEOUT }
  );
  return res.data.response.trim();
}

export async function generateWithImage(prompt: string, imageBase64: string): Promise<string> {
  const res = await axios.post<OllamaResponse>(
    `${OLLAMA_BASE}/api/generate`,
    {
      model: MODEL,
      prompt,
      images: [imageBase64],
      stream: false,
      options: { temperature: 0 },
    },
    { timeout: TIMEOUT }
  );
  return res.data.response.trim();
}

File: backend_py/ollama.py
import os
import httpx

OLLAMA_BASE = os.environ.get("OLLAMA_URL", "http://localhost:11434")
MODEL = "gemma3:4b"
TIMEOUT = 60.0


async def generate(prompt: str) -> str:
    async with httpx.AsyncClient(timeout=TIMEOUT) as client:
        res = await client.post(
            f"{OLLAMA_BASE}/api/generate",
            json={"model": MODEL, "prompt": prompt, "stream": False, "options": {"temperature": 0}},
        )
        res.raise_for_status()
        return res.json()["response"].strip()


async def generate_with_image(prompt: str, image_base64: str) -> str:
    async with httpx.AsyncClient(timeout=TIMEOUT) as client:
        res = await client.post(
            f"{OLLAMA_BASE}/api/generate",
            json={
                "model": MODEL,
                "prompt": prompt,
                "images": [image_base64],
                "stream": False,
                "options": {"temperature": 0},
            },
        )
        res.raise_for_status()
        return res.json()["response"].strip()
